Applications/CLI/ogs_embedded_python.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <pybind11/embed.h>
+#include <logog/include/logog.hpp>
+
+#include "ogs_embedded_python.h"
+
+#include "ProcessLib/BoundaryCondition/Python/PythonBoundaryConditionModule.h"
+
+PYBIND11_EMBEDDED_MODULE(OpenGeoSys, m)
+{
+    DBUG("Binding Python module OpenGeoSys.");
+
+    ProcessLib::pythonBindBoundaryCondition(m);
+}
+
+#ifndef OGS_BUILD_SHARED_LIBS
+
+// Hackish trick that hopefully ensures that the linker won't strip the symbol
+// pointed to by p from the library being built.
+template <typename T>
+void mark_used(T p)
+{
+    volatile T vp = (volatile T)p;
+    vp = vp;
+}
+
+#endif  // OGS_BUILD_SHARED_LIBS
+
+namespace ApplicationsLib
+{
+pybind11::scoped_interpreter setupEmbeddedPython()
+{
+#ifndef OGS_BUILD_SHARED_LIBS
+    // pybind11_init_impl_OpenGeoSys is the function initializing the embedded
+    // OpenGeoSys Python module. The name is generated by pybind11. If it is not
+    // obvious that this symbol is actually used, the linker might remove it
+    // under certain circumstances.
+    mark_used(&pybind11_init_impl_OpenGeoSys);
```

Comment:
just an idea...
Maybe declaring the `pybind11_init_impl_OpenGeoSys` function with `__attribute__((used))` helps (at least for gcc and clang).

Reply:
`__attribute__(used)` is gcc-specific. I didn't find the same for MSVC:
https://msdn.microsoft.com/de-de/library/74f4886t.aspx

Maybe `__declspec(dllexport)` might help?
https://msdn.microsoft.com/de-de/library/3y1sfaz2.aspx
But is that also applicable for static libs?

Anyway, we'd have to distinguish between different compilers, which is also kind of hackish...